### PR TITLE
Revamp Encoder articles

### PR DESCRIPTION
**Pull request by sciencewhiz**
Seperate out form factor of encoder from electrical characteristics. 
Document Duty Cycle encoders and move Analog Encoders from Potentiometer articles to encoder articles.
Remove encoder examples that are better covered elsewhere

### CODE REVIEW

File: source/docs/hardware/sensors/encoders-hardware.rst [hunk 1]
```diff
@@ -5,36 +5,40 @@ Encoders - Hardware
...
 
 These encoders vary in how they are mounted to the mechanism in question.  In addition to these types of encoders, many FRC mechanisms come with quadrature encoders integrated into their design.
 
+There are also three main ways the encoder data is communicated that are typically used in FRC:
+
+- `Quadrature encoders`_
+- `Duty Cycle encoders`_
+- `Analog encoders`_
+
+.. note:: some encoders may support more then one communication method
```
Comment (Starlight220) [inline source/docs/hardware/sensors/encoders-hardware.rst:27]: Capitalization

File: source/docs/hardware/sensors/encoders-hardware.rst [hunk 2]
```diff
@@ -52,19 +62,51 @@ Magnetic encoders
...
   :width: 400
 
-Magnetic encoders require no mechanical coupling to the shaft at all; rather, they track the orientation of a magnet fixed to the shaft.  The `CTRE Mag Encoder <https://store.ctr-electronics.com/srx-mag-encoder/>`__ is a popular option, with many FRC products offering built-in mounting options for it.  While the no-contact nature of magnetic encoders can be handy, they often require precise construction in order to ensure that the magnet is positioned correctly with respect to the encoder.
+Magnetic encoders require no mechanical coupling to the shaft at all; rather, they track the orientation of a magnet fixed to the shaft.  While the no-contact nature of magnetic encoders can be handy, they often require precise construction in order to ensure that the magnet is positioned correctly with respect to the encoder.
+
+Examples of magnetic encoders:
+
+- `CTRE Mag Encoder <https://store.ctr-electronics.com/srx-mag-encoder/>`__
+- `Thrifty Absolute Magnetic Encoder <https://www.thethriftybot.com/bearings/Thrifty-Absolute-Magnetic-Encoder-p421607500>`__
+- `Team 221 Lamprey2 <https://www.andymark.com/products/lamprey-absolute-encoder>`__`
```
Comment (Starlight220) [inline source/docs/hardware/sensors/encoders-hardware.rst:71]:
```suggestion
- `Team 221 Lamprey2 <https://www.andymark.com/products/lamprey-absolute-encoder>`__
```

File: source/docs/hardware/sensors/encoders-hardware.rst [hunk 3]
```diff
@@ -77,3 +119,31 @@ As encoders measure rotation with digital pulses, the accuracy of the measuremen
 Thus, a resolution stated in edges per revolution has a value four times that of the same resolution stated in cycles per revolution.
 
 In general, the resolution of your encoder in edges-per-revolution should be somewhat finer than your smallest acceptable error in positioning.  Thus, if you want to know the mechanism plus-or-minus one degree, you should have an encoder with a resolution somewhat higher than 360 edges per revolution.
+
+Duty Cycle encoders
```
Comment (Starlight220) [inline source/docs/hardware/sensors/encoders-hardware.rst:123]:
```suggestion
Duty Cycle Encoders
```

File: source/docs/software/hardware-apis/sensors/encoders-software.rst [hunk 4]
```diff
@@ -5,25 +5,35 @@ Encoders - Software
 
 .. note:: This section covers encoders in software.  For a hardware guide to encoders, see :ref:`docs/hardware/sensors/encoders-hardware:Encoders - Hardware`.
 
+Encoders are devices used to measure motion (usually, the rotation of a shaft).
+
+.. important:: The classes in this document are only used for encoders that are plugged directly into the roboRIO! Please reference the appropriate vendors' documentation for using encoders plugged into motor controllers.
+
+Quadrature Encoders - The `Encoder`` class
```
Comment (Starlight220) [inline source/docs/software/hardware-apis/sensors/encoders-software.rst:12]:
```suggestion
Quadrature Encoders - The ``Encoder`` class
```

File: source/docs/software/hardware-apis/sensors/encoders-software.rst [hunk 5]
```diff
@@ -115,15 +125,15 @@ The :code:`Encoder` class offers a number of configuration methods:
...
-Reading information from encoders
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Reading information from quadrature encoders
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 The :code:`Encoder` class provides a wealth of information to the user about the motion of the encoder.
 
-Distance
-~~~~~~~~
+Quadrature Distance
```
Comment (Starlight220) [inline source/docs/software/hardware-apis/sensors/encoders-software.rst:133]: I'm not sure the adjective applies well here.

File: source/docs/software/hardware-apis/sensors/encoders-software.rst [hunk 6]
```diff
@@ -226,15 +236,223 @@ To reset an encoder to a distance reading of zero, call the :code:`reset()` meth
...
+.. note:: The :code:`DutyCycleEncoder` class does not make any assumptions about units of distance; it will return values in whatever units were used to calculate the distance-per-rotation value.  Users thus have complete control over the distance units used.
+
+The :code:`DutyCycleEncoder` class offers a number of configuration methods:
+
+.. tabs::
+
+    .. code-tab:: java
+
+        // Configures the encoder to return a distance of 4 for every rotation
+        encoder.setDistancePerRotation(4.);
```
Comment (Starlight220) [inline source/docs/software/hardware-apis/sensors/encoders-software.rst:282]: Don't omit trailing zeroes.

File: source/docs/software/hardware-apis/sensors/encoders-software.rst [hunk 7]
```diff
@@ -226,15 +236,223 @@ To reset an encoder to a distance reading of zero, call the :code:`reset()` meth
...
+
+    .. code-tab:: c++
+
+        // Configures the encoder to return a distance of 4 for every rotation
+        encoder.SetDistancePerRotation(4.);
```
Comment (Starlight220) [inline source/docs/software/hardware-apis/sensors/encoders-software.rst:287]: This method exists?

File: source/docs/software/hardware-apis/sensors/encoders-software.rst [hunk 8]
```diff
@@ -226,15 +236,223 @@ To reset an encoder to a distance reading of zero, call the :code:`reset()` meth
...
+.. note:: The :code:`AnalogEncoder` class does not make any assumptions about units of distance; it will return values in whatever units were used to calculate the distance-per-rotation value.  Users thus have complete control over the distance units used.
+
+The :code:`AnalogEncoder` class offers a number of configuration methods:
+
+.. tabs::
+
+    .. code-tab:: java
+
+        // Configures the encoder to return a distance of 4 for every rotation
+        encoder.setDistancePerRotation(4.);
```
Comment (Starlight220) [inline source/docs/software/hardware-apis/sensors/encoders-software.rst:383]: This method exists?
Again, don't omit the trailing zero.

File: source/docs/software/hardware-apis/sensors/encoders-software.rst [hunk 9]
```diff
@@ -226,15 +236,223 @@ To reset an encoder to a distance reading of zero, call the :code:`reset()` meth
...
+
+    .. code-tab:: c++
+
+        // Configures the encoder to return a distance of 4 for every rotation
+        encoder.SetDistancePerRotation(4.);
+
+Reading information from analog encoders
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The :code:`AnalogEncoder` class provides a wealth of information to the user about the motion of the encoder.
```
Comment (Starlight220) [inline source/docs/software/hardware-apis/sensors/encoders-software.rst:393]: I'm not sure I'd call it a "wealth" of information, but ok

**Review by Starlight220 — CHANGES_REQUESTED** [review]
Can some of the code samples be replaced with RLIs? The `DutyCycleEncoder` example should have relevant code snippets.

File: source/docs/software/hardware-apis/sensors/encoders-software.rst (hunk 5 above)
Comment (sciencewhiz) [reply source/docs/software/hardware-apis/sensors/encoders-software.rst:133]: Need unique labels

File: source/docs/software/hardware-apis/sensors/encoders-software.rst (hunk 7 above)
Comment (sciencewhiz) [reply source/docs/software/hardware-apis/sensors/encoders-software.rst:287]: https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/wpilibj/AnalogEncoder.html#setDistancePerRotation(double)

File: source/docs/software/hardware-apis/sensors/encoders-software.rst (hunk 5 above)
Comment (Starlight220) [reply source/docs/software/hardware-apis/sensors/encoders-software.rst:133]: "Reading distance from X encoder"?

**Comment by sciencewhiz** [comment]
RLIing from DutyCycleEncoder example would break the parallel structure of the 3 encoder types

**Review by Starlight220 — APPROVED** [review]
I'm not convinced that code here can't be RLId, but that can be dealt with later.

**Pull request merged**